### PR TITLE
PRL-5178: Add notification for respondent post SOA

### PR DESCRIPTION
**Pull request by hcarslaw**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-5178

### Change description ###
Adds a notification to the CA respondent dashboard after service of application has been completed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/steps/common/task-list/components/notification-banner/utils.ts [hunk 1]
```diff
@@ -116,3 +122,11 @@ export const isApplicantLIPServingRespondent = (caseData: Partial<CaseWithId>):
 export const isPrimaryApplicant = (caseData: Partial<CaseWithId>, userDetails: UserDetails): boolean => {
   return caseData.applicants?.[0].value.user.idamId === userDetails.id;
 };
+
+export const isPartyServed = (
+  caseData: Partial<CaseWithId>,
+  userDetails: UserDetails,
+  partyType: PartyType
+): boolean => {
+  return getCurrentPartyId(caseData, userDetails, partyType) === caseData.citizenApplicationPacks?.[0].partyId;
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/notification-banner/utils.ts:131]: u need to check if citizenApplicationPacks has any item before you call getCurrentPartyId

File: src/main/steps/prl-cases/dashboard/utils.ts [hunk 2]
```diff
@@ -30,3 +31,36 @@ export const getCasePartyType = (caseData: Partial<CaseWithId>, userId: string):
   }
   return partyType;
 };
+
+export const getCurrentPartyId = (
```
Comment (vivek-sekhar) [inline src/main/steps/prl-cases/dashboard/utils.ts:35]: change to getPartyId

Comment (vivek-sekhar) [reply src/main/steps/prl-cases/dashboard/utils.ts:35]: I think try to add partyId prop to getPartyDetails method itself, which will nt hv the need for this method

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/common/task-list/components/notification-banner/utils.ts [hunk 3]
```diff
@@ -116,3 +122,10 @@ export const isApplicantLIPServingRespondent = (caseData: Partial<CaseWithId>):
 export const isPrimaryApplicant = (caseData: Partial<CaseWithId>, userDetails: UserDetails): boolean => {
   return caseData.applicants?.[0].value.user.idamId === userDetails.id;
 };
+
+export const isPartyServed = (caseData: Partial<CaseWithId>, userDetails: UserDetails): boolean => {
+  return (
+    caseData.citizenApplicationPacks?.length !== 0 &&
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/notification-banner/utils.ts:128]: caseData.citizenApplicationPacks &&  caseData.citizenApplicationPacks.length &&  getPartyDetails(caseData as CaseWithId, userDetails.id)?.id === caseData.citizenApplicationPacks[0].partyId

File: src/main/steps/common/task-list/components/notification-banner/utils.ts [hunk 4]
```diff
@@ -124,8 +124,9 @@ export const isPrimaryApplicant = (caseData: Partial<CaseWithId>, userDetails: U
 };
 
 export const isPartyServed = (caseData: Partial<CaseWithId>, userDetails: UserDetails): boolean => {
-  return (
+  return !!(
+    caseData.citizenApplicationPacks &&
     caseData.citizenApplicationPacks?.length !== 0 &&
-    getPartyDetails(caseData as CaseWithId, userDetails.id)?.id === caseData.citizenApplicationPacks?.[0].partyId
+    getPartyDetails(caseData as CaseWithId, userDetails.id)?.partyId === caseData.citizenApplicationPacks?.[0].partyId
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/notification-banner/utils.ts:130]: you dont have to check length !==0 & not have ? for [0] in applicationPacks

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/common/task-list/components/notification-banner/utils.ts (hunk 4 above)
Comment (hcarslaw) [reply src/main/steps/common/task-list/components/notification-banner/utils.ts]: Replaced check for caseData.citizenApplicationPacks with ? in caseData.citizenApplicationPacks?.length as there was a sonar issue

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**